dom-event-delegation/main.js

```diff
@@ -0,0 +1,16 @@
+var taskList = document.querySelector('.task-list');
+var taskDelegation = document.querySelector('ul');
+
+taskList.addEventListener('click', task);
+taskDelegation.addEventListener('click', task);
+
+function task(event) {
+  if (event.target && event.target.tagName === 'BUTTON') {
+    var element = document.querySelector('li');
+    var ancestor = element.closest('.task-list-item');
+    element.remove('button');
```

Comment:
Two things:
1. `remove` does not have any arguments
2. You are deleting the first `li` on the page, not the one returned from the `closest` method call. If you tried to delete any entry other than the first one, this wouldn't work.